Built in test runner

ryanpdwyer opened the pull request:
Add a built-in test runner so that users can run tests using,

```python
import freqdemod
freqdemod.test()
```

after installing.

JohnMarohn commented:
Ryan, 

I like your proposed changes a lot!

I often install someone's homebrew package and there is no way to test it easily, short of cloning their repository, assuming (*big if*) they even wrote unit tests.   Your proposed modifications give someone who does not want to clone the package a way to easily verify the installation after a ``pip install freqdemod``.  

Before I merge your pull request, can you also please

* Modify HISTORY.rst, making note of your revisions.

* Add a few sentences to the README file.

These additions should only take a minute. 

Thanks,
JM

ryanpdwyer commented:
I added to the history and changed the README to refer the built-in test runner and `unittest`. I renamed testsuite and test methods so they aren't discovered discovered as tests, which could have caused the tests to be run twice in some situations.

JohnMarohn commented:
Looks great!  I am now going to merge your pull request.   Thanks for thinking of this.
